gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -557,6 +582,26 @@ protected void setup(Context context) {
       }
     }
 
+    private JobState readJobState(Context context) throws IOException {
+      JobState jobState = new JobState();
```

Comment:
You can actually change the member variable to `private final JobState jobState = new JobState()` and then just deserialize into `this.jobState`.